scripts/SANS/isis_instrument.py

```diff
@@ -1725,14 +1725,25 @@ def is_run_new_style_run(workspace_ref):
         @param workspace_ref:: A handle to the workspace
         '''
         try:
-            run_num = workspace_ref.getRun().getLogData('run_number').value
+            run_num = LARMOR.get_run_number_from_workspace_reference(workspace_ref)
         except:
+            ws_name = workspace_ref.name()
             run_num = int(re.findall(r'\d+', str(ws_name))[-1])
         if int(run_num) >= 2217:
             return True
         else:
             return False
 
+    @staticmethod
+    def get_run_number_from_workspace_reference(ws_ref):
```

Comment:
Interesting function. I'm sure that something like this can be useful in more places. This reminds me of the exceptions that one needs to make for workspace groups in validateInputs and validation of workspaces in general.